pyfr/backends/base/makoutil.py

```diff
@@ -14,6 +15,11 @@ def ndrange(context, *args):
     return util.ndrange(*args)
 
 
+def ilog2range(context, x):
+    n = x.bit_length() - 1
+    return [int(x/(2**(i + 1))) for i in range(n)]
```

Comment:
In [8]: x = 256

In [9]: [2**i for i in range(x.bit_length())]
Out[9]: [1, 2, 4, 8, 16, 32, 64, 128, 256]